test/unit/org/apache/cassandra/cql3/validation/miscellaneous/SSTablesIteratedTest.java

```diff
@@ -983,4 +988,129 @@ public void testCompactTableWithMultipleRegularColumnsAndColumnDeletion() throws
+
+        executeAndCheck("SELECT * FROM %s WHERE pk = 2", 3, row(2, null, 2, null));
+        executeAndCheck("SELECT * FROM %s WHERE pk = 2 AND c = 1", 3);
+        executeAndCheck("SELECT v FROM %s WHERE pk = 2 AND c = 1", 3);
+
+        executeAndCheck("SELECT * FROM %s WHERE pk = 3", 3, row(3, null, 3, null));
+        executeAndCheck("SELECT * FROM %s WHERE pk = 3 AND c = 1", 2);
+        executeAndCheck("SELECT v FROM %s WHERE pk = 3 AND c = 1", 2);
+    }
+
+    @Test
+    public void testNonCompactTableWithStaticColumn() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int, c int, s int static, v int, PRIMARY KEY(pk, c))");
+
+        execute("INSERT INTO %s (pk, c, s, v) VALUES (?, ?, ?, ?) USING TIMESTAMP 1000", 1, 1, 1, 1);
+        execute("INSERT INTO %s (pk, c, v) VALUES (?, ?, ?) USING TIMESTAMP 1001", 2, 1, 1);
+        execute("INSERT INTO %s (pk, c, v) VALUES (?, ?, ?) USING TIMESTAMP 1002", 3, 1, 1);
+        flush();
+        execute("UPDATE %s USING TIMESTAMP 2000 SET v = ? WHERE pk = ? AND c = ?", 2, 1, 1);
+        execute("UPDATE %s USING TIMESTAMP 2001 SET v = ? WHERE pk = ? AND c = ?", 2, 2, 1);
+        execute("INSERT INTO %s (pk, c, v) VALUES (?, ?, ?) USING TIMESTAMP 2002", 3, 1, 2);
+        flush();
+        execute("UPDATE %s USING TIMESTAMP 3000 SET s = ? WHERE pk = ?", 2, 1);
+        execute("INSERT INTO %s (pk, s) VALUES (?, ?) USING TIMESTAMP 3001", 2, 1);
+        execute("INSERT INTO %s (pk, c, s, v) VALUES (?, ?, ?, ?) USING TIMESTAMP 3002", 3, 1, 1, 3);
+        flush();
+
+        executeAndCheck("SELECT * FROM %s WHERE pk = 1", 3, row(1, 1, 2, 2));
+        executeAndCheck("SELECT * FROM %s WHERE pk = 1 AND c = 1", 3, row(1, 1, 2, 2));
+        executeAndCheck("SELECT v FROM %s WHERE pk = 1 AND c = 1", 3, row(2));
+        executeAndCheck("SELECT s FROM %s WHERE pk = 1", 3, row(2));
+        executeAndCheck("SELECT DISTINCT s FROM %s WHERE pk = 1", 3, row(2));
+
+        executeAndCheck("SELECT * FROM %s WHERE pk = 2", 3, row(2, 1, 1, 2));
+        executeAndCheck("SELECT * FROM %s WHERE pk = 2 AND c = 1", 3, row(2, 1, 1, 2));
+        executeAndCheck("SELECT v FROM %s WHERE pk = 2 AND c = 1", 3, row(2));
+        executeAndCheck("SELECT s FROM %s WHERE pk = 2", 3, row(1));
+        executeAndCheck("SELECT DISTINCT s FROM %s WHERE pk = 2", 3, row(1));
+
+        executeAndCheck("SELECT * FROM %s WHERE pk = 3", 3, row(3, 1, 1, 3));
+        executeAndCheck("SELECT * FROM %s WHERE pk = 3 AND c = 1", 1, row(3, 1, 1, 3));
+        executeAndCheck("SELECT v FROM %s WHERE pk = 3 AND c = 1", 1, row(3));
+        executeAndCheck("SELECT s FROM %s WHERE pk = 3", 3, row(1));
+        executeAndCheck("SELECT DISTINCT s FROM %s WHERE pk = 3", 3, row(1));
+    }
+
+    @Test
+    public void testCompactStaticTable() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int PRIMARY KEY, v int) WITH COMPACT STORAGE");
```

Comment:
The case where we have a compact table but more than one non-key column might be interesting, although scouring this class, it seems like `testCompactAndNonCompactTableWithPartitionTombstones()` sort of already hits it. WDYT?